src/System.Private.CoreLib/shared/System/Buffers/Text/FormattingHelpers.CountDigits.cs

```diff
@@ -103,8 +104,11 @@ public static int CountDigits(uint value)
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
         public static int CountHexDigits(ulong value)
         {
-            // TODO: When x86 intrinsic support comes online, experiment with implementing this using lzcnt.
-            // return 16 - (int)((uint)Lzcnt.LeadingZeroCount(value | 1) >> 3);
+            if (Lzcnt.IsSupported && IntPtr.Size == 8)
+            {
+                var right = 64 - (int)Lzcnt.LeadingZeroCount(value | 1);
+                return (right + 3) >> 2;
+            }
```

Comment:
Nice. Is this going to cause any problems when it's mirrored over to corefx and used by the Utf8Formatter/Parser logic there? Wondering about assembly references and the like.

Reply:
hm.. I am not sure actually probably I need to wrap it with `#if INSIDE_MSCORLIB` or something like that?

Reply:
It should work fine. You may want to copy the file over the corefx locally and build it to verify that there is no ugly surprise.